c/httpfileservice.c

```diff
@@ -197,6 +197,35 @@ void deleteUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void createFileFromUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
+  if (isDir(absolutePath)) {
+      int folderNameLen = strlen(absolutePath);
+      if(folderNameLen == 0 || absolutePath == NULL) {
+         respondWithJsonError(response, "Failed to idenity the folder pointed", 400, "Bad Request");
+      }
+      int slashPos = lastIndexOf(absolutePath, folderNameLen, '/');
+      char *tarFileName = (slashPos == -1) ? absolutePath : absolutePath + slashPos + 1;
+      char finalFileName[strlen(tarFileName) + strlen(".tar")];
+      strcpy(finalFileName, tarFileName);
+      strcat(finalFileName,".tar");
+      char finalCommand[strlen("/bin/sh -c 'tar -cf ") + strlen(finalFileName) + strlen(absolutePath) + 2];
+      strcpy(finalCommand, "/bin/sh -c 'tar -cf ");
```

Comment:
You can define a single variable with the command and in case the command changes you don't have to update several places.
```c
const char *command = "/bin/sh -c 'tar -cf ";
```